### PR TITLE
IOS-530: Delete folder and IOS-529: Delete file

### PR DESCRIPTION
**Pull request by SilviuOdobescu**

### CODE REVIEW

File: AlfrescoApp/Model/Managers/RealmSyncManager.m [hunk 1]
```diff
@@ -205,23 +205,81 @@ - (void)disableSyncForAccountFromConfig:(UserAccount *)account
...
+        });
+    }
+}
+
+- (void)handleDocumentForDelete:(AlfrescoNode *)document arrayOfNodesToDelete:(NSMutableArray *)arrayToDelete arrayOfNodesToSaveLocally:(NSMutableArray *)arrayToSave arrayOfPaths:(NSMutableArray *)arrayOfPaths inRealm:(RLMRealm *)realm
+{
+    SyncNodeStatus *syncNodeStatus = [self.syncHelper syncNodeStatusObjectForNodeWithId:[self.syncHelper syncIdentifierForNode:document] inSyncNodesStatus:self.syncNodesStatus];
+    syncNodeStatus.totalSize = 0;
+    RealmSyncNodeInfo *nodeInfo = [[RealmManager sharedManager] syncNodeInfoForObjectWithId:[self.syncHelper syncIdentifierForNode:document] ifNotExistsCreateNew:NO inRealm:realm];
+    [arrayToDelete addObject:nodeInfo];
```
Comment (alexp84) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: Please perform a nil check before adding nodeInfo to the array.

File: AlfrescoApp/Model/Managers/RealmSyncManager.m [hunk 2]
```diff
@@ -205,23 +205,81 @@ - (void)disableSyncForAccountFromConfig:(UserAccount *)account
...
+    BOOL isModifiedLocally = [self isNodeModifiedSinceLastDownload:document inRealm:realm];
+    if(isModifiedLocally)
+    {
+        [arrayToSave addObject:document];
+    }
+    
+    NSString *nodeSyncName = [self.syncHelper syncNameForNode:document inRealm:realm];
+    NSString *syncNodeContentPath = [[self.syncHelper syncContentDirectoryPathForAccountWithId:[AccountManager sharedManager].selectedAccount.accountIdentifier] stringByAppendingPathComponent:nodeSyncName];
+    [arrayOfPaths addObject:syncNodeContentPath];
```
Comment (alexp84) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: It's a very long shot, but syncNodeContentPath could be nil.

File: AlfrescoApp/Model/Managers/RealmSyncManager.m [hunk 3]
```diff
@@ -205,23 +205,81 @@ - (void)disableSyncForAccountFromConfig:(UserAccount *)account
...
+}
+
+- (void)handleFolderForDelete:(AlfrescoNode *)folder arrayOfNodesToDelete:(NSMutableArray *)arrayToDelete arrayOfNodesToSaveLocally:(NSMutableArray *)arrayToSave arrayOfPaths:(NSMutableArray *)arrayOfPaths inRealm:(RLMRealm *)realm
+{
+    RealmSyncNodeInfo *folderInfo = [[RealmManager sharedManager] syncNodeInfoForObjectWithId:[self.syncHelper syncIdentifierForNode:folder] ifNotExistsCreateNew:NO inRealm:realm];
+    [arrayToDelete addObject:folderInfo];
```
Comment (alexp84) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: Please perform a nil check before adding folderInfo to the array.

File: AlfrescoApp/Model/Managers/RealmSyncManager.m (hunk 1 above)
Comment (SilviuOdobescu) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: There is a check being made in the FileFolderCollectionViewController that makes sure that this object will never be nil (it checks that the node is in the sync list, thus present in DB). Will add the check for nil here as well as a best practice.

File: AlfrescoApp/Model/Managers/RealmSyncManager.m (hunk 2 above)
Comment (SilviuOdobescu) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: See the comment above.

File: AlfrescoApp/Model/Managers/RealmSyncManager.m (hunk 3 above)
Comment (SilviuOdobescu) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: See the comment above

**Pull request merged**